react-native/ios/RealmReact/RealmReact.mm

```diff
@@ -55,20 +56,21 @@ - (JSGlobalContextRef)ctx;
     id rctJSContext = object_getIvar(executor, contextIvar);
     if (!rctJSContext && create) {
         Class RCTJavaScriptContext = NSClassFromString(@"RCTJavaScriptContext");
-        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:)];
-        assert(signature);
 
-        // React Native 0.14+ expects a JSContext here, but we also support 0.13.x, which takes a JSGlobalContextRef.
-        if (!strcmp([signature getArgumentTypeAtIndex:2], "@")) {
-            JSContext *context = [[JSContext alloc] init];
-            rctJSContext = [[RCTJavaScriptContext alloc] initWithJSContext:(void *)context];
+        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:onThread:)];
+        if (signature) {
+            // for RN 0.28.0+
+            rctJSContext = [executor context];
```

Comment:
@appden - last time I tried it `RCTJavaScriptContextCreatedNotification` is posted async and in some cases would arrive after RN tried to call our apis which obviously wouldn't work. Probably worth trying this out again since the RN code base has changed significantly since. The `JSContext` hack sounds terrible but I think your right about its reliability. Only issue is it might be too reliable - ie how do we prevent other `JSContext`s outside of RN from being initialized with our apis?